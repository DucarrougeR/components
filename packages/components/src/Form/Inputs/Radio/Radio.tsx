import styled from 'styled-components'
import { reset, space, SpaceProps } from '@looker/design-tokens'
import { InputProps } from '../InputProps'

export interface RadioProps extends SpaceProps, Omit<InputProps, 'type'> {}

export const Radio = styled.input.attrs({ type: 'radio' })<RadioProps>`
  ${reset}
  ${space}
`
