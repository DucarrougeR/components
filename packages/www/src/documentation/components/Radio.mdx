---
name: Radio
menu: Form
route: /form/radio
---

import { ComponentDetail } from '../../components'

# `Radio`

<ComponentDetail
  of="Radio"
  status="experimental"
  figma="file/2MG6DoSjk4IaLnjjFCnKFf/Forms"
  github="Form/Inputs/Radio/Radio"
/>

## Default Radio Button

The `<Radio />` component renders a single radio button on the page, with no accompanying label. It is generally used to construct higher-order components like the `<FieldRadio />`. If you are building a form, you probably want to use `<FieldRadio />` instead as it provides label support.

import { Radio } from '@looker/components'
import { Playground } from 'docz'

<Playground>
  <Radio />
</Playground>

## Name and ID

A name and ID can be specified in the `<Radio />` component. Radio buttons of the same `name` are considered to be in a 'radio group' and only one radio button in a radio group can be selected at a time.

<Playground>
  <Radio name="someName" id="one" />
  <br />
  <Radio name="someName" id="two" />
  <br />
  <Radio name="someName" id="three" />
</Playground>

## Checked Property

The `<Radio />` component accepts the `checked` property, which either sets the radio button as selected or not.

<Playground>
  <Radio checked={true} readOnly />
  <br />
  <Radio checked={false} readOnly />
</Playground>

## Disabled Property

Use the disable property to gray out the radio button, making it unclickable.

<Playground>
  <Radio disabled />
  <br />
  <Radio disabled checked />
</Playground>
