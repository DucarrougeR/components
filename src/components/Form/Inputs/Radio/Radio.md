```js noeditor
const StatusAndResources = require('../../../../../styleguide_components/StatusAndResources')
  .StatusAndResources

;<StatusAndResources
  status="experimental"
  figmaURL="https://www.figma.com/file/2MG6DoSjk4IaLnjjFCnKFf/Forms"
  githubURL="https://github.com/looker/lens/blob/master/src/components/Form/Inputs/Radio.tsx"
  feedbackTitle="Radio Component Feedback"
/>
```

### Default Radio Button

The `<Radio />` component renders a single radio button on the page, with no accompanying label. It is generally used to construct higher-order components like the `<FieldRadio />`. If you are building a form, you probably want to use `<FieldRadio />` instead as it provides label support.

```js
import { Radio } from './Radio'
;<Radio />
```

### Name and ID

A name and ID can be specified in the `<Radio />` component. Radio buttons of the same `name` are considered to be in a 'radio group' and only one radio button in a radio group can be selected at a time.

```js
import { Radio } from './Radio'
;<>
  <Radio name="someName" id="one" />
  <br />
  <Radio name="someName" id="two" />
  <br />
  <Radio name="someName" id="three" />
</>
```

### Checked Property

The `<Radio />` component accepts the `checked` property, which either sets the radio button as selected or not.

```js
import { Radio } from './Radio'
;<>
  <Radio checked={true} readOnly />
  <br />
  <Radio checked={false} readOnly />
</>
```

### Disabled Property

Use the disable property to gray out the radio button, making it unclickable.

```js
import { Radio } from './Radio'
;<>
  <Radio disabled />
  <br />
  <Radio disabled checked />
</>
```
