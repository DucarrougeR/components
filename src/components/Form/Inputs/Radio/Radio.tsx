import * as React from 'react'
import { styled } from '../../../../style'
import { Box, BoxProps } from '../../../Box'
import { InputProps } from '../InputProps'

export interface RadioProps extends BoxProps<HTMLInputElement>, InputProps {
  /**
   * Determines if the radio button is selected or not.
   */
  checked?: boolean
}

const InternalRadio: React.FC<RadioProps> = ({ validationType, ...props }) => {
  const type = { type: 'radio' }
  return <Box is="input" {...props} {...type} />
}

const RadioFactory = React.forwardRef((props: RadioProps, ref) => (
  <InternalRadio innerRef={ref} {...props} />
))

export const Radio = styled<RadioProps>(RadioFactory)``
