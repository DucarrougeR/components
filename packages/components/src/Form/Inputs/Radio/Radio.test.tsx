import 'jest-styled-components'
import * as React from 'react'
import { mountWithTheme } from '@looker/components-test-utils'
import { assertSnapshot } from '@looker/components-test-utils'
import { Radio } from './Radio'

test('Radio default', () => {
  assertSnapshot(<Radio />)
})

test('Radio checked set to true', () => {
  assertSnapshot(<Radio checked={true} />)
})

test('Radio checked set to false', () => {
  assertSnapshot(<Radio checked={false} />)
})

test('Radio with name and id', () => {
  assertSnapshot(<Radio name="Chuck" id="Chucky" />)
})

test('Radio should accept disabled', () => {
  assertSnapshot(<Radio disabled />)
})

test('Radio with aria-describedby', () => {
  assertSnapshot(<Radio aria-describedby="some-id" />)
})

test('Should trigger onChange handler', () => {
  let counter = 0
  const handleChange = () => counter++

  const wrapper = mountWithTheme(<Radio onChange={handleChange} />)

  wrapper.find('input').simulate('change', { target: { value: '' } })
  expect(counter).toEqual(1)
})
