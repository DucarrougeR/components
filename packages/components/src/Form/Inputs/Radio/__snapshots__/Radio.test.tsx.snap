// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio checked set to false 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

<input
  checked={false}
  className="c0"
  type="radio"
/>
`;

exports[`Radio checked set to true 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

<input
  checked={true}
  className="c0"
  type="radio"
/>
`;

exports[`Radio default 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

<input
  className="c0"
  type="radio"
/>
`;

exports[`Radio should accept disabled 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

<input
  className="c0"
  disabled={true}
  type="radio"
/>
`;

exports[`Radio with aria-describedby 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

<input
  aria-describedby="some-id"
  className="c0"
  type="radio"
/>
`;

exports[`Radio with name and id 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

<input
  className="c0"
  id="Chucky"
  name="Chuck"
  type="radio"
/>
`;
